rwengine/src/render/OpenGLRenderer.hpp

```diff
@@ -252,21 +252,21 @@ class OpenGLRenderer : public Renderer {
 
     std::string getIDString() const override;
 
-    ShaderProgram* createShader(const std::string& vert,
+    std::unique_ptr<ShaderProgram> createShader(const std::string& vert,
                                 const std::string& frag) override;
-    void setProgramBlockBinding(ShaderProgram* p, const std::string& name,
+    void setProgramBlockBinding(std::unique_ptr<ShaderProgram>& p, const std::string& name,
```

Comment:
I think it would make more sense for the ``setProgramBlockBinding``, ``setUniform`` and ``useProgram`` to take either a ``ShaderProgram*`` or a ``ShaderProgram&``, the interface shouldn't care if they're in a unique_ptr or not, and they shouldn't be null anyway.